x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/RegionMapChart/ChoroplethMap.tsx

```diff
@@ -0,0 +1,276 @@
+
+      // cleanup function called when component unmounts
+      return () => {
+        canvasElement.removeEventListener('wheel', controlScrollZoomOnWheel);
+      };
+    }
+  }, [controlScrollZoomOnWheel]);
+
+  // side effect swaps the old handlers with new ones when they update
+  useEffect(() => {
+    if (map) {
+      if (
+        updateHoverStateOnMousemoveRef.current &&
+        updateHoverStateOnMouseoutRef.current
+      ) {
+        map.off('mousemove', updateHoverStateOnMousemoveRef.current);
+        map.off('mouseout', updateHoverStateOnMouseoutRef.current);
+      }
+      map.on('mousemove', updateHoverStateOnMousemove);
+      map.on('mouseout', updateHoverStateOnMouseout);
+      updateHoverStateOnMousemoveRef.current = updateHoverStateOnMousemove;
+      updateHoverStateOnMouseoutRef.current = updateHoverStateOnMouseout;
+    }
+  }, [map, updateHoverStateOnMousemove, updateHoverStateOnMouseout]);
+
+  // side effect replaces choropleth layer with new one on data changes
+  useEffect(() => {
+    if (map) {
+      // find first symbol layer to place new layer in correct order
+      const symbolLayer = (map.getStyle().layers || []).find(
+        ({ type }) => type === 'symbol'
+      );
+      if (map.getLayer(CHOROPLETH_LAYER_ID)) {
+        map.removeLayer(CHOROPLETH_LAYER_ID);
+      }
+      if (data.length) {
+        map.addLayer(
+          {
+            id: CHOROPLETH_LAYER_ID,
+            type: 'fill',
+            source: CHOROPLETH_POLYGONS_SOURCE_ID,
+            layout: {},
+            paint: {
+              'fill-opacity': 0.75,
+              'fill-color': data.length
+                ? {
+                    property: geojsonKeyProperty,
+                    stops: data.map(({ key, value }) => [
+                      key,
+                      getProgressionColor(getValueScale(value))
+                    ]),
+                    type: 'categorical',
+                    default: 'transparent'
+                  }
+                : 'transparent'
+            }
+          },
+          symbolLayer ? symbolLayer.id : undefined
+        );
+      }
+    }
+  }, [map, data, getValueScale, geojsonKeyProperty]);
+
+  // side effect to only render the Popup when hovering a region with data
+  useEffect(() => {
+    if (popupContainerRef.current && map && popupRef.current) {
+      if (hoverState.geojsonProperties && hoverState.data) {
+        popupRef.current.setDOMContent(popupContainerRef.current).addTo(map);
+        if (popupContainerRef.current.parentElement) {
+          popupContainerRef.current.parentElement.style.pointerEvents = 'none';
+        }
+      } else {
+        popupRef.current.remove();
+      }
+    }
+  }, [map, hoverState]);
+
+  // render map container and tooltip in a hidden container
+  return (
+    <>
+      <div ref={containerRef} style={{ height: 256, ...style }} />
+      <div style={{ display: 'none' }}>
+        <div ref={popupContainerRef}>
+          <ToolTip
```

Comment:
Why is this renamed from `renderToolTip` to `ToolTip`? And why is `RegionMapChartToolTip` passed in as a prop in the first place? What about extract `RegionMapChartToolTip` to a separate file, and then use it directly?

Reply:
> I can definitely extract `RegionMapChartToolTip` to a separate file. I didn't at first because it's so small and specific. I like to keep component definitions of this nature very close to where they will be used. in the same source file even. But I'm not opposed to moving it.

Yeah, I think it makes sense to move it to it's own file and import it instead of passing it as a prop. Afaict we don't need to pass in different tooltips. If we eventually need it it'll be very easy to add.

Reply:
> Is a specific version of Chrome where you are seeing no background?

I'm on "Version 76.0.3809.100 (Official Build) (64-bit)"

Reply:
I'm still seeing this issue and @dgieselaar mentioned it in here: 
https://github.com/elastic/kibana/pull/43567#issuecomment-524542851

I think this is a blocker.